src/parser/mod.rs

```diff
@@ -20,10 +20,25 @@ pub type ParseResult<T, I> = nom::IResult<T, I, Error>;
 
 pub struct Parser;
 
+#[macro_export]
+macro_rules! jinko_ex {
+    ($($t:tt) *) => {
+        $crate::Parser::parse(stringify!( $( $t ) * )).unwrap().execute().unwrap();
+    }
+}
+
 #[macro_export]
 macro_rules! jinko {
     ($($t:tt) *) => {
-        $crate::Parser::parse(stringify!( $( $t ) * )).unwrap().execute();
+        {
+            let mut ctx = Context::new();
+            let (_, insts) = $crate::parser::Construct::many_instructions(stringify!( $( $t ) * )).unwrap();
+            for inst in insts {
+                inst.execute(&mut ctx);
```

Comment:
Since you execute everything here, do you really need to call `ctx.execute().unwrap()` afterwards (in my previous comment)? Am I missing something?